fiftyone/utils/eval/detection.py

```diff
@@ -176,45 +183,50 @@ def evaluate_detections(
     processing_frames = samples._is_frame_field(pred_field)
     save = eval_key is not None
 
-    if save:
-        tp_field = "%s_tp" % eval_key
-        fp_field = "%s_fp" % eval_key
-        fn_field = "%s_fn" % eval_key
-
     if config.requires_additional_fields:
         _samples = samples
     else:
         _samples = samples.select_fields([gt_field, pred_field])
 
+    # Determine number of workers
+    if num_workers is None:
+        num_workers = max(1, mp.cpu_count() - 1)
+
+    logger.info(
+        f"Evaluating detections in parallel with {num_workers} workers..."
+    )
+
     matches = []
-    logger.info("Evaluating detections...")
-    for sample in _samples.iter_samples(progress=progress, autosave=save):
-        if processing_frames:
-            docs = sample.frames.values()
-        else:
-            docs = [sample]
-
-        sample_tp = 0
-        sample_fp = 0
-        sample_fn = 0
-        for doc in docs:
-            doc_matches = eval_method.evaluate(doc, eval_key=eval_key)
-            matches.extend(doc_matches)
-            tp, fp, fn = _tally_matches(doc_matches)
-            sample_tp += tp
-            sample_fp += fp
-            sample_fn += fn
-
-            if processing_frames and save:
-                doc[tp_field] = tp
-                doc[fp_field] = fp
-                doc[fn_field] = fn
-
-        if save:
-            sample[tp_field] = sample_tp
-            sample[fp_field] = sample_fp
-            sample[fn_field] = sample_fn
+    # Create a thread pool
+    with ThreadPoolExecutor(max_workers=num_workers) as executor:
+        futures = []
+
+        # Submit samples to the worker pool
+        for sample in _samples.iter_samples(
+            progress=progress, batch_size=batch_size, autosave=save
+        ):
+            if processing_frames:
+                docs = sample.frames.values()
+            else:
+                docs = [sample]
+
+            future = executor.submit(
+                _process_sample, docs, eval_method, eval_key, processing_frames
+            )
+            futures.append((future, sample))
+
+        # Collect results
+        for future, sample in futures:
```

Comment:
We probably should define a constant somewhere so thaat we can collect the results earlier without having to maintain a large list